lib/combinat.gi

```diff
@@ -2446,6 +2446,72 @@ InstallGlobalFunction( "IteratorOfPartitions", function( n )
     end );
 
 
+#############################################################################
+##
+#F  IteratorOfPartitionsSet( <set> )
+##
+##  all unordered partitions of the set <A>set</A> into pairwise disjoint
+##  nonempty sets.
+##
+InstallGlobalFunction( IteratorOfPartitionsSet, function( s )
+    local nextIterator, isDone, m, out, i, shallowCopy;
+
+      if not IsSet(s) then
+          Error( "<s> must be a set" );
+      fi;
+
+    nextIterator := function(iter)
+      local j, max, part, m, out, i;
+      if iter!.s = [] then
+        iter!.next := false;
+        return [];
+      fi;
+      part := StructuralCopy(iter!.next);
+      j := Size(iter!.next);
+        # Compute next restricted growth string using a_i \leq Max({a_0 .. a_{i-1}})+1
+      while j > 1 do
+        max := Maximum(iter!.next{[1 .. j-1]});
+        if iter!.next[j] <= max then
+          iter!.next[j] := iter!.next[j]+1;
+            # Covert restricted growth string to partition of set
+          m:=Maximum(part)+1;
+          out := List([1 .. m], t -> []);;
+          for i in [1 .. Size(part)] do
+            Add(out[part[i]+1], iter!.s[i]);
+          od;
+          return out;;
+        else
+          iter!.next[j] := 0;
+          j := j-1;
+        fi;
+      od;
+      m:=Maximum(part)+1;
+      out := List([1 .. m], t -> []);;
+      for i in [1 .. Size(part)] do
+        Add(out[part[i]+1], iter!.s[i]);
+      od;
+      iter!.next := false;
+      return out;;
+    end;
+
+    shallowCopy := function( iter )
+        return rec(
+            next := ShallowCopy( iter!.next ) );
+    end;
```

Comment:
```suggestion
    shallowCopy := function(iter)
      return rec( next := ShallowCopy( iter!.next ) );
    end;
```
for consistency with the rest of your code... Though I'd actually prefer shorthand notation (also for `isDone`), but that's subjective and not a requirement:
```suggestion
    shallowCopy := iter -> rec( next := ShallowCopy( iter!.next ) );
```

Reply:
I have gone with the shorthand notation for both shallowCopy and isDone